corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -189,12 +189,16 @@ def send_to_elasticsearch(index_info, doc_type, doc_id, es_getter, name, data=No
             if delete:
                 es_interface.delete_doc(alias, doc_type, doc_id)
             else:
-                params = {'retry_on_conflict': 2}
                 if es_merge_update:
-                    es_interface.update_doc_fields(alias, doc_type, doc_id, fields=data, params=params)
+                    # The `retry_on_conflict` param is only valid on `update`
+                    # requests. ES <5.x was lenient of its presence on `index`
+                    # requests, ES >=5.x is not.
```

Comment:
but just to triple check, in 2.x it's presence or absence has no effect on anything other than an update request?

Reply:
Correct. In fact I went deep into the git history to figure out why we're even adding it on `index` requests and discovered that the current state is a regression bug. In previous implementations we were only adding it on `update` requests, but that detail got lost in a refactor (possibly during the ES7 changes).

Reply:
Awesome, thanks for digging!